### PR TITLE
Merge sort

### PR DESCRIPTION
**Pull request by cahudson94**
Pull for submission

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -271,3 +271,12 @@ All modules and tests located in src directory.
...
         - if more then one type of input is given return error message.
+
+
+## Merge Sort
+- module: merge_sort.py
+- test: test_merge_sort.py
+- public methods:
+    ### merge_sort:
+        - sort the input list.
+        - if more then one type of input is given return error message.
```
Comment (clair3st) [inline README.md]: You need to add Big O time complexity for this.

File: src/working/merge_sort.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+            while len(right) > ri:
+                data[di] = right[ri]
+                ri += 1
+                di += 1
+
+        return data
+
+    except TypeError:
+        return 'This sort method only accepts values of one \
```
Comment (clair3st) [inline src/working/merge_sort.py]: Are you trying to raise an error here? This is just returning a string which is a little weird.

File: src/working/merge_sort.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+            di = 0
+            li = 0
+            ri = 0
+
+            while len(left) > li and len(right) > ri:
+                if not isinstance(left[li], int):
+                    raise TypeError()
+                if not isinstance(right[ri], int):
+                    raise TypeError()
```
Comment (clair3st) [inline src/working/merge_sort.py]: It would be more helpful to your users if they had a message with the Error raised.

File: src/working/test_merge_sort.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+
+TEST_PARAMS = [
+    ([i for i in range(50)]),
+    ([randint(1, 500) for i in range(50)]),
+    ([5, 9, 1, 2, 6])
+]
+
+
+PARAMS_TABLE = [
```
Comment (clair3st) [inline src/working/test_merge_sort.py]: I would like to see you add some more test cases with duplicate numbers and negative numbers as input.
